src/rosdiscover/observer/observer.py

```diff
@@ -59,3 +60,20 @@ def observe(self) -> SystemSummary:
             An immutable representation of the system architecture.
         """
         ...
+
+    @abstractmethod
+    def execute_script(self, path_on_host: str) -> Popen:
+        """Executes a script on the executing container.
+
+        Parameters
+        ----------
+        path_on_host: str
+            The path to the script on the host (commands in the script should
+            be relative to the container, not the host).
+
+        Returns
+        -------
+        int
```

Comment:
doesn't match return type

Reply:
also, this method probably doesn't need to be abstract. I think that you could getaway with the same method for ROS1 and ROS2?

Reply:
Agreed. Done.